xen/drivers/passthrough/arm/ipmmu-vmsa.c

```diff
@@ -1143,6 +1149,14 @@ static int ipmmu_attach_device(struct iommu_domain *io_domain,
 #if 0
 		ret = ipmmu_domain_init_context(domain);
 #endif
+		/*
+		 * Here we have to disable IPMMU TLB cache function of IPMMU caches
+		 * that do require such action.
+		 */
+		if (domain->mmus[0]->is_mmu_tlb_disabled)
+			ipmmu_ctx_write1(domain, IMSCTLR,
+					ipmmu_ctx_read(domain, IMSCTLR) | 0xE0000000);
```

Comment:
`0xE0000000` is the some magic value.

Reply:
Yes, the code of mmngr [1] I used as a reference, operated with the same magic value
[1] https://github.com/renesas-rcar/mmngr_drv/commit/4aed49e7577a41152977dd99d4392adf1d9beed2
Do you want me to define it somehow?

Reply:
Yes, I think it should be defined. Renesas code is a bad excuse. I'm sure, it will not pass code review in LKML.

Reply:
Don't mind